### PR TITLE
Cmd line parser

### PR DESCRIPTION
**Pull request by tvdu94**
Adding the parsing of a line, to detect options. Only with Mermaid for the moment

### CODE REVIEW

File: src/main/java/com/github/forax/umldoc/editor/MermaidCmdLineParser.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.Optional;
+import picocli.CommandLine;
+import picocli.CommandLine.Option;
+
+/**
+ * This class is used to parse the command line for the mermaid export.
+ *
+ */
+public class MermaidCmdLineParser implements CommandLineParser {
+  private boolean hasGenerationStarted = false;
```
Comment (forax) [inline src/main/java/com/github/forax/umldoc/editor/MermaidCmdLineParser.java:13]: It's not necessary, the code that calls that interface already know if it is inside of outside the %%umldoc block

File: src/main/java/module-info.java [hunk 2]
```diff
@@ -1,3 +1,5 @@
 module com.github.forax.umldoc {
   requires org.objectweb.asm;
+  requires info.picocli;
+  exports com.github.forax.umldoc.editor;
```
Comment (forax) [inline src/main/java/module-info.java:4]: `exports`or `opens` ?

**Comment by forax** [comment]
And argvester [1] is far better than picoli because it supports records :)

[1] https://github.com/forax/argvester

**Comment by tvdu94** [comment]
> And argvester [1] is far better than picoli because it supports records :)
> 
> [1] https://github.com/forax/argvester

I used picocli according to the SPEC.md file. Do you still want me to change to argvester?

**Comment by tvelu77** [comment]
the failed test seems to come from picocli dependency.
ModuleScrapper tests that fail are :
[ERROR]   ModuleScrapperTest.scrapCore:46 » NullPointer
[ERROR]   ModuleScrapperTest.scrapModule:29 » NullPointer
We are checking how to fix these errors

**Comment by tvelu77** [comment]
It is now fixed with Esteban's recent pull request

**Pull request merged**